<!DOCTYPE html>
<html>
  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <link href='https://fonts.googleapis.com/css?family=Chivo:900' rel='stylesheet' type='text/css'>
    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/github-dark.css" media="screen">
    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <title>Workshop-popgenome by tonig-evo</title>
  </head>

  <body>
    <div id="container">
      <div class="inner">

        <header>
          <h1>Workshop-popgenome</h1>
          <h2>Tutorial about R package PopGenome</h2>
        </header>

        <section id="downloads" class="clearfix">
          <a href="https://github.com/tonig-evo/workshop-popgenome/zipball/master" id="download-zip" class="button"><span>Download .zip</span></a>
          <a href="https://github.com/tonig-evo/workshop-popgenome/tarball/master" id="download-tar-gz" class="button"><span>Download .tar.gz</span></a>
          <a href="https://github.com/tonig-evo/workshop-popgenome" id="view-on-github" class="button"><span>View on GitHub</span></a>
        </section>

        <hr>

        <section id="main_content">
          <h1>
<a id="introduction" class="anchor" href="#introduction" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Introduction</h1>

<p>PopGenome is an R package
(<a href="https://cran.r-project.org/web/packages/PopGenome/index.html">https://cran.r-project.org/web/packages/PopGenome/index.html</a>) for
analyses of population genomic data \citep{Pfeifer2014}. For this
tutorial, please make sure that your R working directory is set
correctly and you have all the packages installed, e.g.
install.packages("PopGenome"). The following files are necessary to
conduct this practical session:</p>

<ul>
<li><p>fasta_file.txt, fasta files for one locus from different
<em>Arabidopsis thaliana</em> individuals (accessions) and the outgroup
sequence from <em>Arabidopsis lyrata</em> in the folder <strong>fasta</strong></p></li>
<li><p>LGE22.gff, LGE22.vcf, LGE22.fa in subfolders in the folder
<strong>great_tit</strong></p></li>
<li><p>variants.vcf, ind_species1.txt, ind_species2.txt, rad_assembly.fa
in the folder <strong>rad</strong></p></li>
</ul>

<p>To get an overview about the file contents, inspect files with a text
Editor (e.g. Notepad+) or via the command line (e.g. more) before
starting to work on them.</p>

<h1>
<a id="load-library-into-r" class="anchor" href="#load-library-into-r" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Load library into R</h1>

<p># Loading module library(PopGenome)</p>

<h1>
<a id="fasta-files" class="anchor" href="#fasta-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Fasta Files</h1>

<h2>
<a id="reading-fasta-files" class="anchor" href="#reading-fasta-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Reading Fasta Files</h2>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Reading data, read fasta from folder </span>
<span class="pl-smi">GENOME.class</span> <span class="pl-k">&lt;-</span> readData(<span class="pl-s"><span class="pl-pds">"</span>fasta<span class="pl-pds">"</span></span>) 
get.sum.data(<span class="pl-smi">GENOME.class</span>)</pre></div>

<p>What statistics can one obtain from get.sum.data function?</p>

<p>Folders <strong>fasta_a</strong>, <strong>fasta_b</strong>, <strong>fasta_c</strong> contain modified
alignments. Identify the differences between the datasets. Why PopGenome
fails to load fasta files?</p>

<h2>
<a id="obtaining-summary-statistics-from-alignments" class="anchor" href="#obtaining-summary-statistics-from-alignments" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Obtaining summary statistics from alignments</h2>

<p><strong>Note:</strong> Since calculation of certain population genetic parameter can
be computational intense, they have to be executed separately
beforehand. For this modules have to be run. Note that module <strong>Fst</strong>
has to be executed with <strong>F_st</strong>. The statistic Tajima’s D is part of
the module <strong>neutrality</strong> not <strong>Fst</strong></p>

<div class="highlight highlight-source-r"><pre><span class="pl-c"># Available statistics and examples </span>
show.slots(<span class="pl-smi">GENOME.class</span>) 
<span class="pl-c"># Run</span>
<span class="pl-smi">necessary</span> <span class="pl-smi">module</span> <span class="pl-smi">GENOME.class</span> \<span class="pl-k">&lt;-</span> <span class="pl-c1">F</span><span class="pl-k">~</span><span class="pl-smi">S</span><span class="pl-k">~</span><span class="pl-c1">T</span>.stats(<span class="pl-smi">GENOME.class</span>) <span class="pl-smi">GENOME.class</span>
\<span class="pl-k">&lt;-</span> neutrality.stats(<span class="pl-smi">GENOME.class</span>) <span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">n.sites</span> <span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">Pi</span>
<span class="pl-smi">GENOME.class</span><span class="pl-k">@</span><span class="pl-smi">TajimaD</span></pre></div>

<p>What different modules are available? (show.slots)</p>

<p>What module is necessary to be executed in order to obtain Wall.B
\citep{Wall1999}?</p>

<p>How could one obtain a per site estimate of Pi?</p>

<h2>
<a id="obtaining-statistics-for-regions" class="anchor" href="#obtaining-statistics-for-regions" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Obtaining statistics for regions</h2>

<p># Available region data and statistics <a href="mailto:GENOME.class@region.data">GENOME.class@region.data</a>
<a href="mailto:GENOME.class@region.stats">GENOME.class@region.stats</a> # Examples
GENOME.class@<a href="mailto:region.data@biallelic.sites">region.data@biallelic.sites</a>[[1]][1:10]
GENOME.class@region.data@transitions[[1]][1:10]</p>

<p>How many sites have gaps?</p>

<p>How many singletons are in the dataset?</p>

<p>What is the difference between <em>region.data</em> and <em>region.stats</em>?\
(see also <strong>Whole_genome_analyses_using_VCF_files.pdf</strong>)</p>

<h2>
<a id="define-outgroups-and-populations" class="anchor" href="#define-outgroups-and-populations" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Define outgroups and populations</h2>

<p><strong>Note:</strong> If one ore more outgroup sequences are defined, PopGenome will
only consider SNPs where the outgroup is monomorphic; the monomorphic
nucleotide is then automatically defined as the major allele (encoded by
0).</p>

<p>get.individuals(GENOME.class) GENOME.class &lt;-
set.populations(GENOME.class,list(
c("CON","KAS-1","RUB-1","PER-1","RI-0","MR-0","TUL-0"),
c("MH-0","YO-0","ITA-0","CVI-0","COL-2","LA-0","NC-1") )) # Check
whether grouping is set correctly GENOME.class@region.data@populations
GENOME.class@region.data@populations2 GENOME.class@region.data@outgroup
# Recalculate statistics for populations GENOME.class &lt;-
neutrality.stats(GENOME.class) <a href="mailto:GENOME.class@Tajima.D">GENOME.class@Tajima.D</a></p>

<p>Name three implemented statistics that require an outgroup.</p>

<p>What do you have to pay attention to when applying the McDonald-Kreitman
test?\
(see <strong>Whole_genome_analyses_using_VCF_files.pdf</strong>)</p>

<h2>
<a id="including-features-from-gff-files-to-fasta-files" class="anchor" href="#including-features-from-gff-files-to-fasta-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Including features from GFF files to Fasta files</h2>

<ul>
<li><p>If no gff-file was specified when the data was read in, it is
assumed that the alignment is in the correct reading frame (starting
at a first codon position)</p></li>
<li><p>The GFF folder contains GFF-files for each alignment stored in the
FASTA folder. The GFF files should have the same names (without any
extensions like .fas or .gff) as the corresponding FASTA files to
ensure that sequence and annotation are matched correctly.</p></li>
</ul>

<h1>
<a id="analysing-vcf-files-for-whole-genome-data" class="anchor" href="#analysing-vcf-files-for-whole-genome-data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Analysing VCF files for whole genome data</h1>

<h2>
<a id="loading-vcf-files" class="anchor" href="#loading-vcf-files" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Loading VCF files</h2>

<p>There are two ways to read in VCF files:</p>

<ul>
<li><p>Folder of VCF files: <strong>readData</strong></p></li>
<li><p>Single VCF file: <strong>readVCF</strong></p></li>
</ul>

<p>To read a VCF file using <em>readVCF</em> it needs to be compressed with
<em>bgzip</em> and indexed with <em>tabix</em>. The tabix files need to be placed in
the same folder as the vcf file.</p>

<p># What parameters need to be defined GENOME.class&lt;-
readVCF("great~t~it/vcf/LGE22.vcf.gz",
6000,"chrLGE22~P~arus~M~ajor~b~uild~1~.0.2",1,773534)
<a href="mailto:GENOME.class@region.names">GENOME.class@region.names</a> GENOME.class&lt;- neutrality.stats(GENOME.class,
FAST=TRUE) get.sum.data(GENOME.class) <a href="mailto:GENOME.class@region.data">GENOME.class@region.data</a></p>

<p>What parameters need to be defined to <em>readVCF</em>? (see <strong>PopGenome.pdf</strong>)</p>

<p>What is the overall diversity (theta and pi) of chromosome LGE22?</p>

<h2>
<a id="loading-vcf-files-with-annotation" class="anchor" href="#loading-vcf-files-with-annotation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Loading VCF files with annotation</h2>

<p>GENOME2.class &lt;- readData("great~t~it/vcf2",format="VCF",
gffpath="great~t~it/gff") get.sum.data(GENOME2.class)
<a href="mailto:GENOME2.class@region.data">GENOME2.class@region.data</a> #get.codons(GENOME.class,1) #split &lt;-
splitting.data(GENOME2.class, subsites="CDS") GENOME2.class &lt;-
set.synnonsyn(GENOME2.class, ref.chr="great~t~it/fasta/LGE22.fasta")
GENOME2.class@region.data@synonymous
GENOME2.class@region.data@CodingSNPS</p>

<p>What are the advantages and disadvantages of <em>readData</em> and <em>readVCF</em>?\
(see <strong>Whole_genome_analyses_using_VCF_files.pdf</strong>)</p>

<p>What is the overall diversity (theta and pi) of chromosome LGE22 for
synonymous and nonsynonymous sites?</p>

<h1>
<a id="analysing-radseq-data-using-vcf-" class="anchor" href="#analysing-radseq-data-using-vcf-" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Analysing RADseq data using VCF </h1>

<p># SPLIT VCF FILE
VCF~s~plit~i~nto~s~caffolds("rad/variants.vcf","rad~s~plit~v~cf") #
READ IN DATA GENOME.class &lt;- readData("rad~s~plit~v~cf",format="VCF")
pop1 &lt;-as.character(read.table("rad/ind~s~pecies1.txt")[[1]]) pop2
&lt;-as.character(read.table("rad/ind~s~pecies2.txt")[[1]]) GENOME.class
&lt;- set.populations(GENOME.class,list(pop1,pop2),diploid=TRUE) # CHECK
GENOME.class@populations</p>

<h2>
<a id="obtaining-statistics-from-multiple-vcfs-derived-from-radseq" class="anchor" href="#obtaining-statistics-from-multiple-vcfs-derived-from-radseq" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Obtaining statistics from multiple VCFs derived from RADseq</h2>

<p># NEUTRALITY STATISTICS GENOME.class &lt;- neutrality.stats(GENOME.class,
FAST=TRUE) get.neutrality(GENOME.class)[[1]] <a href="mailto:GENOME.class@Tajima.D">GENOME.class@Tajima.D</a> #
FST GENOME.class &lt;- F~S~T.stats(GENOME.class)
get.F~S~T(GENOME.class)[[1]] <a href="mailto:GENOME.class@nucleotide.F">GENOME.class@nucleotide.F</a>~S~T # DIVERSITY
GENOME.class &lt;- diversity.stats(GENOME.class)
get.diversity(GENOME.class) <a href="mailto:GENOME.class@nuc.diversity.within">GENOME.class@nuc.diversity.within</a> # SFS
GENOME.class &lt;-
detail.stats(GENOME.class,site.spectrum=TRUE,site.FST=TRUE) results &lt;-
get.detail(GENOME.class) GENOME.class@<a href="mailto:region.stats@minor.allele.freqs">region.stats@minor.allele.freqs</a></p>

<p>Plot a site-frequency-spectrum for each population.</p>

<p>results &lt;- get.detail(GENOME.class) allele~F~reqs &lt;-
GENOME.class@<a href="mailto:region.stats@minor.allele.freqs">region.stats@minor.allele.freqs</a>[[1]]</p>

<p>freq.table &lt;- list() freq.table[[1]] &lt;- table(allele~F~reqs) sfs &lt;-
data.frame(freq.table)</p>

<p>library(ggplot2) ggplot(sfs, aes(x=allele~F~reqs, y=Freq)) +
geom~b~ar(stat = ’identity’)</p>

<h1>
<a id="more-information" class="anchor" href="#more-information" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>More information</h1>

<p>More information are available in three pdfs accompanied by the program
(see folder <strong>pdf</strong>):</p>

<ol>
<li><p><strong>An introduction to the PopGenome package</strong>: Sliding window
analysis, reading SNP data files, coalescent simulations</p></li>
<li><p><strong>Whole genome analyses using PopGenome and VCF files</strong>: Details
about reading tabixed VCF files, examples, graphical output,
parallel read-in, pre-filtering VCF files, WhopGenome
\citep{Wittelsbuerger2015}</p></li>
<li><p><strong>Package PopGenome</strong>: Documentation about all implemented functions
with examples</p></li>
</ol>
        </section>

        <footer>
          Workshop-popgenome is maintained by <a href="https://github.com/tonig-evo">tonig-evo</a><br>
          This page was generated by <a href="https://pages.github.com">GitHub Pages</a>. Tactile theme by <a href="https://twitter.com/jasonlong">Jason Long</a>.
        </footer>

        
      </div>
    </div>
  </body>
</html>
